### ISSUE TITLE
increase linting timeout

### ISSUE THREAD

**Issue by syphar** [LambdaLint/lambdalint.github.io#6]
for our definitely not small application, the linters timeout. 

Mostly still showing "running linter" 
<img width="247" alt="Bildschirmfoto 2019-12-19 um 07 56 05" src="https://user-images.githubusercontent.com/540890/71151842-0c1c1700-2235-11ea-83e8-bbe8fc247682.png">

Sometimes I also saw a "timeout after 200s" message. 

Or is the linter stuck here? 
(without logs difficult to debug)

it happens not only with one, but more linters
<img width="338" alt="Bildschirmfoto 2019-12-19 um 07 58 14" src="https://user-images.githubusercontent.com/540890/71151939-57362a00-2235-11ea-9286-5c9d0dc873c0.png">

**Comment by codingjoe** [comment]
Hi @syphar,
Yes, the current timeout for subprocess that runs the linter itself is set to 200s:
https://github.com/FussyFox/lintipy/blob/5c89ef330f178d6ae854072d6b871de29b28ccb0/lintipy.py#L155

This is mainly due to AWS Lambda's 300s function timeout. This the total and must also facilitate pulling the code and setting sending status updates to GitHub, like the timeout message you are seeing.

Anyhow, [apparently AWS increased the limit a function can be run](https://aws.amazon.com/de/about-aws/whats-new/2018/10/aws-lambda-supports-functions-that-can-run-up-to-15-minutes/). I would be ok, increasing the execution time, since my AWS bill doesn't kill me yet.

That being said, we should still verify that there isn't another issue, that is causing the timeout, except this probably being a very large repository (let me guess which one 😉 ). Can you provide me with the commit hash and a repo name I'll look though the logs. I can also share them, if you want me too.

Merry Xmas 🎄 
-Johannes

**Comment by syphar** [comment]
@codingjoe yep, this one :) 

Thermondo/thermondo-backend
5366a2a524b7aa7563b70f6b172ec7496ff22f58

just a thought, the archive-link you are using to download, does that include the whole history? 
(which in this case is even bigger)

**Comment by syphar** [comment]
just checked, archive is only the code

**Comment by codingjoe** [comment]
Soo, I removed you comment, since the repo name doesn't need to be publicly available.

Anyhow, I looked through the code and the timeout occurred during the execution of the linter. That being said, the code download takes a whopping 8 seconds followed by a 10 second gzip extract.

I did only look into the `isort`, but actually the timeout message you see at GitHub will be different depending on what part of the process times out. So my guess is, it is always the linter. Knowing the repo, that doesn't come as a surprise... it's just too much my code 😜

Going forward, there are two options.
1. Use GitHub actions. They are slower, but you have the added benefit of adding constraints. e.G. running your test suite only after the linters have passed. If you are looking for a large example of a GitHub CI workflow with Docker, PostGis and other extras, [here is something I wrote recently](https://github.com/codingjoe/django/blob/30962/flake8/.github/workflows/ci.yml).
2. We – and by we I mean you – increase the timeout to 15min. I am happy to do that, but only if it's worth the effort. How long does `isort` or `falke8` take on your local machine as a single process `-j0`? You can increase the limit in the line that I pointed out earlier. Once the package is released, dependabot will update all functions. However, the function timeout will also be increased here:
https://github.com/FussyFox/isort/blob/03ec6cc9bf5c88539818020f394566ed9c120a00/.travis.yml#L21

Your call :)
